tests/baselines/reference/fixingTypeParametersRepeatedly1.types

```diff
@@ -35,7 +35,7 @@ declare function g();
 >g : { <T>(x: T, y: (p: T) => T, z: (p: T) => T): T; (): any; }
 
 g("", x => null, x => x.toLowerCase());
->g("", x => null, x => x.toLowerCase()) : any
+>g("", x => null, x => x.toLowerCase()) : string

```

Comment:
Like many thing reference sin the OP, this is just a straight improvment - previously `any` would get fixed to the second parameter's parameter during overload checking early on, now that gets discarded and we get the type from the overload we actually pick - `string`.